common/beerocks/bcl/unit_tests/wireless_utils_test.cpp

```diff
@@ -13,120 +13,265 @@
+    // clang-format on
 
-    EXPECT_EQ(get_operating_class_by_channel_helper(104, 40), 123);
-    EXPECT_EQ(get_operating_class_by_channel_helper(144, 40), 123);
+    std::string result = "bandwidth_";
 
-    EXPECT_EQ(get_operating_class_by_channel_helper(149, 20), 124);
-    EXPECT_EQ(get_operating_class_by_channel_helper(161, 20), 124);
+    auto bandwidth = static_cast<beerocks::eWiFiBandwidth>(value);
 
-    EXPECT_EQ(get_operating_class_by_channel_helper(165, 20), 125);
-    EXPECT_EQ(get_operating_class_by_channel_helper(169, 20), 125);
+    const auto &it = bandwidths.find(bandwidth);
+    if (bandwidths.end() == it) {
+        result += "invalid_" + std::to_string(value);
+    } else {
+        result += it->second;
+    }
 
-    EXPECT_EQ(get_operating_class_by_channel_helper(149, 40), 126);
-    EXPECT_EQ(get_operating_class_by_channel_helper(157, 40), 126);
+    return result;
+}
 
-    EXPECT_EQ(get_operating_class_by_channel_helper(153, 40), 127);
-    EXPECT_EQ(get_operating_class_by_channel_helper(161, 40), 127);
+/**
+ * @brief Converts a operating class code to its string representation.
+ *
+ * Helper function to build the operating class part of the name of a value-parameterized test
+ * parameter.
+ *
+ * @param value Operating class code.
+ * @return Operating class as a string.
+ */
+std::string operating_class_to_string(uint8_t value)
+{
+    return "operating_class_" + std::to_string(value);
+}
 
-    /* For 80MHz+, we use starting freq but the table uses center freq */
-    EXPECT_EQ(get_operating_class_by_channel_helper(36, 80), 128);
-    EXPECT_EQ(get_operating_class_by_channel_helper(52, 80), 128);
-    EXPECT_EQ(get_operating_class_by_channel_helper(100, 80), 128);
-    EXPECT_EQ(get_operating_class_by_channel_helper(116, 80), 128);
-    EXPECT_EQ(get_operating_class_by_channel_helper(132, 80), 128);
-    EXPECT_EQ(get_operating_class_by_channel_helper(149, 80), 128);
+/**
+ * Fixture class to write value-parameterized tests for method
+ * wireless_utils::get_operating_class_by_channel()
+ *
+ * See Value-Parameterized Tests documentation at
+ * https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#value-parameterized-tests
+ */
+class WirelessUtilsGetOperatingClassByChannelTest
+    : public testing::TestWithParam<tGetOperatingClassByChannelParam> {
+};
 
-    EXPECT_EQ(get_operating_class_by_channel_helper(36, 160), 129);
-    EXPECT_EQ(get_operating_class_by_channel_helper(100, 160), 129);
+/**
+ * @brief Converts a parameter of a get_operating_class_by_channel() value-parameterized test to
+ * its string representation.
+ *
+ * Helper function to create a custom test name suffix based on the test parameter value.
+ *
+ * @param param value-parameterized test parameter.
+ * @return Value-parameterized test parameter as a string.
+ */
+std::string operating_class_by_channel_param_to_string(
+    const testing::TestParamInfo<WirelessUtilsGetOperatingClassByChannelTest::ParamType> &info)
```

Comment:
Isn't this just
```suggestion
    const testing::TestParamInfo<tGetOperatingClassByChannelParam> &info)
```
?

Reply:
Yes it is. Fixed